baselines/fedmix/fedmix/dataset_preparation.py

```diff
@@ -0,0 +1,157 @@
+"""..."""
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+import torchvision.transforms as transforms
+from torch.utils.data import Subset
+from torchvision.datasets import CIFAR10, CIFAR100
+
+
+def _download_cifar10():
+    """..."""
+    transform = transforms.Compose(
+        [
+            transforms.ToTensor(),
+            transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
+        ]
+    )
+
+    trainset = CIFAR10("./dataset", train=True, download=True, transform=transform)
+    testset = CIFAR10("./dataset", train=False, download=True, transform=transform)
+    return trainset, testset
+
+
+def _download_cifar100():
+    """..."""
+    transform = transforms.Compose(
+        [
+            transforms.ToTensor(),
+            transforms.Normalize((0.5071, 0.4867, 0.4408), (0.2675, 0.2565, 0.2761)),
+        ]
+    )
+
+    trainset = CIFAR100("./dataset", train=True, download=True, transform=transform)
+    testset = CIFAR100("./dataset", train=False, download=True, transform=transform)
+    return trainset, testset
+
+
+def _partition_trainset(
```

Comment:
This function does return one partition per client, and each partition does contain only data points from the desired number of classes. But these datapoints are not exclusive to each client, some images are present in several partitions.

Reply:
Will change this and let you know the results.